programs/account-compression/src/instructions/update_address_merkle_tree.rs

```diff
@@ -25,57 +24,63 @@ pub fn process_update_address_merkle_tree<'info>(
     ctx: Context<'_, '_, '_, 'info, UpdateMerkleTree<'info>>,
     // Index of the Merkle tree changelog.
     changelog_index: u16,
-    // Index of the address to dequeue.
-    queue_index: u16,
+    // Address to dequeue.
+    value: [u8; 32],
     // Index of the next address.
-    address_next_index: usize,
+    next_index: usize,
     // Value of the next address.
-    address_next_value: [u8; 32],
+    next_value: [u8; 32],
     // Low address.
-    low_address: RawIndexingElement<usize, 32>,
+    low_address_index: u64,
+    low_address_value: [u8; 32],
+    low_address_next_index: u64,
     // Value of the next address.
     low_address_next_value: [u8; 32],
     // Merkle proof for updating the low address.
-    low_address_proof: [[u8; 32]; 22],
+    low_address_proof: [[u8; 32]; 16],
     // ZK proof for integrity of provided `address_next_index` and
     // `address_next_value`.
     _next_address_proof: [u8; 128],
 ) -> Result<()> {
-    let mut address_queue = ctx.accounts.queue.load_mut()?;
-    let address_queue = address_queue_from_bytes_mut(&mut address_queue.queue);
+    let address_queue = ctx.accounts.queue.to_account_info();
+    let mut address_queue = address_queue.try_borrow_mut_data()?;
+    let mut address_queue = unsafe { address_queue_from_bytes_zero_copy_mut(&mut address_queue)? };
+
     let mut merkle_tree = ctx.accounts.merkle_tree.load_mut()?;
 
-    // Remove the address from the queue.
-    let address = address_queue
-        .dequeue_at(queue_index)
-        .map_err(|_| AccountCompressionErrorCode::AddressQueueDequeue)?
-        .ok_or(AccountCompressionErrorCode::InvalidIndex)?;
+    let sequence_number = merkle_tree.load_merkle_tree()?.merkle_tree.sequence_number;
+    let value = BigUint::from_bytes_le(value.as_slice());
+
+    // Mark the address with the current sequence number.
+    address_queue
+        .mark_with_sequence_number(&value, sequence_number)
+        .map_err(ProgramError::from)?;
```

Comment:
same thing wrt ``mark_with_sequence_number``as in nullify leaves instruction

Reply:
Done